### ISSUE TITLE
Clicking a link very frequently fails

### ISSUE THREAD

**Issue by Atario** [hexchat/hexchat#1414]
Seems like at least 50% of the time, clicking a link in a channel does nothing at all.  Mostly I have to resort to right-clicking and choosing "Open Link in Browser" to get it to work.  There doesn't seem to be any rhyme or reason to which ones work and which don't, either.

This is v.2.10.2 portable on Win7.

**Comment by Arnavion** [comment]
https://github.com/hexchat/hexchat/issues/363#issuecomment-15821989

**Comment by Jaex** [comment]
While clicking link mouse down and mouse up events must trigger in same cursor position otherwise link not opens. So while clicking if you move cursor one pixel before releasing it then link not opens. And this is very annoying. I have no idea why HexChat handling click event like this.

**Comment by TingPing** [comment]
Fixed by https://github.com/hexchat/hexchat/commit/971e5cf4fe528f145a38dadcf75310b1ea5ae46c

**Comment by OlegYch** [comment]
still fails a lot in 2.12.1
i'm on Windows 8

**Comment by FichteFoll** [comment]
I also experience this occasionally on 2.12.1. It usually happens consecutively and I suspect a program restart to fix it, but I haven't specifically investigated further. Windows 7.

**Comment by Fenrirthviti** [comment]
I imagine this issue is no longer being monitored, but experiencing similar behavior on 2.12.4.

This issue is NOT the "pixel perfect click" issue. Something in the program causes all clicks for all links to fail, no matter how carefully they are clicked. For me, highlighting some text (doesn't matter where) and then clicking off the highlighted text (to deselect it) fixes the issue and links work properly again. I have been unable to replicate this reliably, so no clue as to the actual cause (might be related to some period of inactivity?), but as I use IRC pretty much all day every day, it comes up 4-5 times a day.

Happens on all versions of Windows I use (Win7/8/10), so I don't think this is an OS problem.

**Comment by FichteFoll** [comment]
I mentioned this on IRC a few months ago, but I'm also experiencing this exact same issue under Arch Linux.

**Comment by tiggybitch** [comment]
I use 2.12.4 on Windows 10. Up until the past week or so I have been able to click a URL and have it open with no issues. Yes, gui_url_mod is set to 0. I am back to having to right click and open URLs. 

I have to agree with Fenrirtheviti that this is NOT a "pixel perfect click" issue. I've tried restarting the program only to have the issue persist.

**Comment by Fenrirthviti** [comment]
Definitely still present for me as well. I'm guessing this issue is not monitored due to being closed. I'll look into providing a detailed report with more info the next time it occurs for me.

**Comment by FichteFoll** [comment]
Considering the rarity of the issue, it's could be a race condition. It's also interesting that this happens on Windows *and* Linux, though it's rarer on Linux (at least for me).

@TingPing, would you consider reopening the issue?

**Comment by tiggybitch** [comment]
I restarted Hexchat after rebooting my computer and was able to open 3 links without having to right click. Just went to click a URL and once again it is back to right clicking to open. I have tried in multiple places of the URL and still nothing.

**Comment by Fenrirthviti** [comment]
Is there any further information I can provide to help track this down? I spent a bit of time trying to debug, and couldn't sort out on my own, but I'm happy to provide any debug logs/info possible.

**Comment by melt7777** [comment]
Also affected latest hexchat on windows 10. Thank you.

**Comment by entuland** [comment]
I think it's interfering with the auto-copy feature in the advanced settings.

Disabling such auto-copy seems to fix the program randomly ignoring the clicks for me, probably the code that handles the auto-copy is hijacking the mouse events there.

**Comment by melt7777** [comment]
@entuland Your solution worked for me! I turned that setting off and the links are clickable again. This has bothered me for years. Thank you so much!!!!

**Comment by haarp** [comment]
Nope, this isn't the solution. I've been having this bug for years and never had any auto-copy enabled.

Granted, it only appears after clicking 1 or 2 links.

**Comment by entuland** [comment]
@haarp Yep, there definitely must be something more to it as I still get some clicks ignored here and there, but it's behaving way better now with the auto-copy feature disabled - at least that the case for me.

@melt7777 glad it's working for you!

**Comment by gothicserpent** [comment]
I was having this issue when double-clicking links. If I hover the mouse over a link and single left click it seems to always work (even with auto-copy enabled or disabled).

I think single left clicking links is the solution here.

**Comment by Fenrirthviti** [comment]
I can confirm this still happens, and single left clicking (which is how I have opened links since I started using HexChat) does not work when the bug occurs. It sounds like you are encountering a different issue.

**Comment by gothicserpent** [comment]
No, I'm referring to the same issue. Hexchat 2.14.3 on win10.

**Comment by Fenrirthviti** [comment]
Then I posit again, that whatever issue you're running in to is something else. There is a known bug where clicks need to be pixel perfect between the first and second with double clicking links. When the issue occurs on my end (which still happens 4-5 times daily, and has for almost 7 years now), single clicking does not work. I have basically never used double click to open links in Hexchat. My browser windows are always open on a tertiary monitor, and maximized. While I appreciate you offering workarounds, let's not try to detract from the fact there is a very real and unavoidable issue with links not working consistently in Hexchat. This is the only application I use that has this issue, and the suggestion that some kind of extremely specific workflow and preparation of my environment is required to avoid it is, frankly, ridiculous.

**Comment by madprops** [comment]
Still happens haha

**Comment by davidsedlar** [comment]
Yes still happens quite a bit. Usually have to right click -> Open link in browser for it to work.
